third_party/terraform/resources/resource_cloudtasks_queue.go

```diff
@@ -0,0 +1,342 @@
+		return fmt.Errorf("Error deleting task queue: %s", err)
+	}
+
+	log.Printf("[DEBUG] Finished deleting Task Queue: %#v\n%#v\n", name, resp)
+
+	return nil
+}
+
+func resourceCloudTasksQueueImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("Invalid import id %q. Expecting {project}/{location}/{queue}", d.Id())
+	}
+
+	d.Set("project", parts[0])
+	d.Set("location", parts[1])
+	d.Set("name", parts[2])
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func flattenRateLimits(in *cloudtasks.RateLimits) []interface{} {
+	m := make(map[string]interface{})
+
+	if in != nil {
+		m["max_dispatches_per_second"] = in.MaxDispatchesPerSecond
+		m["max_concurrent_dispatches"] = in.MaxConcurrentDispatches
+		m["max_burst_size"] = in.MaxBurstSize
+	}
+
+	return []interface{}{m}
+}
+
+func expandRateLimits(configured interface{}) *cloudtasks.RateLimits {
+	l := configured.([]interface{})
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	rateLimits := l[0].(map[string]interface{})
+
+	return &cloudtasks.RateLimits{
+		MaxBurstSize:            int64(rateLimits["max_burst_size"].(int)),
+		MaxConcurrentDispatches: int64(rateLimits["max_concurrent_dispatches"].(int)),
+		MaxDispatchesPerSecond:  rateLimits["max_dispatches_per_second"].(float64),
+	}
+}
+
+func flattenRetryConfig(in *cloudtasks.RetryConfig) []interface{} {
+	m := make(map[string]interface{})
+
+	if in != nil {
+		m["max_attempts"] = in.MaxAttempts
+		m["max_doublings"] = in.MaxDoublings
+		m["max_backoff"] = in.MaxBackoff
+		m["min_backoff"] = in.MinBackoff
+	}
+
+	return []interface{}{m}
+}
+
+func expandRetryConfig(configured interface{}) *cloudtasks.RetryConfig {
+	l := configured.([]interface{})
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	retryConfig := l[0].(map[string]interface{})
+
+	return &cloudtasks.RetryConfig{
+		MaxAttempts:  int64(retryConfig["max_attempts"].(int)),
+		MaxDoublings: int64(retryConfig["max_doublings"].(int)),
+		MaxBackoff:   retryConfig["max_backoff"].(string),
+		MinBackoff:   retryConfig["min_backoff"].(string),
+	}
+}
+
+func flattenAppEngineRoutingOverride(project string, configured interface{}, in *cloudtasks.AppEngineRouting) []interface{} {
+	var instance, service, version string
+
+	l := configured.([]interface{})
+	if len(l) != 0 && l[0] != nil {
+		appEngineRouting := l[0].(map[string]interface{})
+		instance = appEngineRouting["instance"].(string)
+		service = appEngineRouting["service"].(string)
+		version = appEngineRouting["version"].(string)
+	}
+
+	m := make(map[string]interface{})
+
+	if in != nil {
+		m["host"] = in.Host
+		m["instance"] = in.Instance
+		m["service"] = in.Service
+		m["version"] = in.Version
+
+		// instance, service, and version aren't returned by the API
+		// so if the host matches expected value, override the returned values
+		if generateHost(project, instance, service, version) == in.Host {
```

Comment:
Is this similar to a DiffSuppressFunc? Why would we do this in the flatten rather than as a DSF